src/services/earn/earn-service.ts

```diff
@@ -978,10 +953,100 @@ export class EarnService implements IEarnService {
       ],
     });
 
-    return swapData;
+    const multicalls = [...previousCalls, ...calls.map(({ tx }) => tx), runSwapData];
+    const multicall = encodeFunctionData({
+      abi: companionAbi,
+      functionName: 'multicall',
+      args: [multicalls],
+    });
+
+    return multicall;
   }
+  private async simulateSwaps({
```

Comment:
I think it would be best if this returned a best quote, or if it threw an error if all quotes failed